content/academy/puppeteer_playwright/executing_scripts.md

```diff
@@ -0,0 +1,155 @@
+---
+title: III - Executing scripts
+description: Understand the two different contexts which your code can be run, and how to run custom scripts in the context of the browser.
+menuWeight: 7.3
+paths:
+    - puppeteer-playwright/executing-scripts
+---
+
+# [](#executing-scripts) Executing scripts
+
+An important concept to understand when dealing with headless browsers is the **context** in which your code is being run. For example, if you try to use the native `fs` Node.js module (used in the previous lesson) while running code in the context of the browser, errors will be thrown saying that it is undefined. Similarly, if you are trying to use `document.querySelector()` or other browser-specific functions in the server-side Node.js context, errors will also be thrown.
+
+Here is an example of a common mistake made by beginners to Puppeteer/Playwright:
+
+```JavaScript
+// THIS IS INCORRECT CODE
+import { chromium } from 'playwright';
+
+const sleep = (secs) => new Promise((r) => setTimeout(r, secs * 1000));
```

Comment:
Get rid of `sleep` everywhere and just use `page.waitForTimeout()` It's a nice opportunity to remind the users of the waitFor functions